spark-connector/src/main/java/com/datastrato/gravitino/spark/plugin/GravitinoDriverPlugin.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.spark.plugin;
+
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.spark.GravitinoSparkConfig;
+import com.datastrato.gravitino.spark.catalog.GravitinoCatalogManager;
+import com.datastrato.gravitino.spark.catalog.GravitinoHiveCatalog;
+import com.google.common.base.Preconditions;
+import java.util.Collections;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.spark.SparkConf;
+import org.apache.spark.SparkContext;
+import org.apache.spark.api.plugin.DriverPlugin;
+import org.apache.spark.api.plugin.PluginContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * GravitinoDriverPlugin creates GravitinoCatalogManager to fetch catalogs from Gravitino and
+ * register Gravitino catalogs to Spark.
+ */
+public class GravitinoDriverPlugin implements DriverPlugin {
+  private static final Logger LOG = LoggerFactory.getLogger(GravitinoDriverPlugin.class);
+
+  private GravitinoCatalogManager catalogManager;
+
+  @Override
+  public Map<String, String> init(SparkContext sc, PluginContext pluginContext) {
+    SparkConf conf = sc.conf();
+    String gravitinoUri = conf.get(GravitinoSparkConfig.GRAVITINO_URI);
+    String metalake = conf.get(GravitinoSparkConfig.GRAVITINO_METALAKE);
+    Preconditions.checkArgument(
+        StringUtils.isNoneBlank(gravitinoUri),
+        String.format(
+            "%s:%s, should not be empty", GravitinoSparkConfig.GRAVITINO_URI, gravitinoUri));
+    Preconditions.checkArgument(
+        StringUtils.isNoneBlank(metalake),
+        String.format(
+            "%s:%s, should not be empty", GravitinoSparkConfig.GRAVITINO_METALAKE, metalake));
+
+    catalogManager = GravitinoCatalogManager.create(gravitinoUri, metalake);
+    catalogManager.loadCatalogsFromGravitino();
+    registerGravitinoCatalogs(conf, catalogManager.getGravitinoCatalogs());
```

Comment:
I would suggest loading and register the catalog when used. It is not necessary to register them all at first, this will have many limitations, if something is changed, you cannot sync it in Spark.

Reply:
Currently, you're using sparksql configuration to register catalog. Maybe you can use the internal API to register programmatically.

Reply:
register programmatically will make load catalog cost more time to retrieve catalog from Gravitino.  For "a.b" , spark will try to load catalog to check if 'a' is a catalog or database.   see: https://github.com/apache/spark/blob/branch-3.4/sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/LookupCatalog.scala#122
```scala
        try {
          Some((catalogManager.catalog(nameParts.head), nameParts.tail.asIdentifier))
        } catch {
          case _: CatalogNotFoundException =>
            Some((currentCatalog, nameParts.asIdentifier))
        }
```